### PR TITLE
feat(CART-CPC-1058): Modify the cart

### PR DESCRIPTION
**Pull request by NtsoaElie**
**JIRA:** link to jira ticket
## Context:
What is the ticket about and why are we doing this change.
## Does this PR change the .vscode folder in petclinic-frontend?:
If the PR changes the .vscode folder, explain why in detail because it should not. 
Be sure to include cgerard321 as a reviewer.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
What are the various changes and what other modules do those changes affect.
This can be bullet point or sentence format.
## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 
## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

**Comment by Drexjar** [comment]
looks good

**Review by Drexjar — APPROVED** [review]
looks good

File: cart-service/src/main/java/com/petclinic/cartsservice/businesslayer/CartServiceImpl.java [hunk 1]
```diff
@@ -37,4 +39,27 @@ public Mono<CartResponseModel> getCartByCartId(String cartId) {
...
+        return cartRepository.findCartByCartId(cartId)
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new NotFoundException("Cart id was not found: " + cartId))))
+                .flatMap(foundCart -> cartRequestModel
+                        .map(EntityModelUtil::toCartEntity)
+                        .doOnNext(c -> c.setCartId(foundCart.getCartId()))
+                        .doOnNext(c -> c.setCustomerId(foundCart.getCustomerId()))
+                        .doOnNext(c -> c.setProductIds(foundCart.getProductIds()))
+                        .doOnNext(c -> c.setId(foundCart.getId()))
+                )
+
```
Comment (Hares-2088) [inline cart-service/src/main/java/com/petclinic/cartsservice/businesslayer/CartServiceImpl.java:54]: what you do here is :
1. find the actual cart
2. map the request to an entity
3. take the data from the found entity and put it in the request that you have transformed into an entity
this means that the update is not going to work

**Review by AlexCristea04 — APPROVED** [review]
modified based on previous comments, looks good now

**Review by Drexjar — APPROVED** [review]
Looks good

**Review by Hares-2088 — APPROVED** [review]
All good. you fixed the logic problem

**Pull request merged**